src/pages/iou/MoneyRequestSelectorPage.js

```diff
@@ -0,0 +1,178 @@
+    ...withCurrentUserPersonalDetailsDefaultProps,
+};
+
+function MoneyRequestSelectorPage(props) {
+    const iouType = useRef(lodashGet(props.route, 'params.iouType', ''));
+    const {translate} = useLocalize();
+
+    const isEditing = useRef(lodashGet(props.route, 'path', '').includes('amount'));
+    const reportID = useRef(lodashGet(props.route, 'params.reportID', ''));
+
+    const title = {
+        [CONST.IOU.MONEY_REQUEST_TYPE.REQUEST]: translate('iou.requestMoney'),
+        [CONST.IOU.MONEY_REQUEST_TYPE.SEND]: translate('iou.sendMoney'),
+        [CONST.IOU.MONEY_REQUEST_TYPE.SPLIT]: translate('iou.splitBill'),
+    };
+    const titleForStep = isEditing.current ? translate('iou.amount') : title[iouType.current];
+
+    const navigateBack = () => {
+        Navigation.goBack(isEditing.current ? ROUTES.getMoneyRequestConfirmationRoute(iouType.current, reportID.current) : null);
+    };
+
+    const selectedTab = props.tabSelected ? props.tabSelected : CONST.TAB.TAB_MANUAL;
+    const [isDraggingOver, setIsDraggingOver] = useState(false);
+
+    return (
+        <FullPageNotFoundView shouldShow={!IOUUtils.isValidMoneyRequestType(iouType.current)}>
+            <ScreenWrapper includeSafeAreaPaddingBottom={false}>
+                {({safeAreaPaddingBottomStyle}) => (
+                    <DragAndDrop
+                        dropZoneId={CONST.RECEIPT.DROP_NATIVE_ID}
+                        activeDropZoneId={CONST.RECEIPT.ACTIVE_DROP_NATIVE_ID}
+                        onDragEnter={() => {
+                            setIsDraggingOver(true);
+                        }}
+                        onDragLeave={() => {
+                            setIsDraggingOver(false);
+                        }}
+                        onDrop={(e) => {
+                            e.preventDefault();
+                            setIsDraggingOver(false);
+                            const file = lodashGet(e, ['dataTransfer', 'files', 0]);
+
+                            if (!AttachmentUtils.isValidFile(file, props)) {
+                                return;
+                            }
+
+                            const filePath = URL.createObjectURL(file);
+                            IOU.setMoneyRequestReceipt(filePath, Str.isImage(file.name));
+                            NavigateToNextIOUPage(props.iou, iouType, reportID, props.report, props.currentUserPersonalDetails);
+                        }}
+                    >
+                        <View
+                            nativeID={CONST.RECEIPT.DROP_NATIVE_ID}
+                            style={[styles.flex1, safeAreaPaddingBottomStyle]}
+                        >
+                            <HeaderWithBackButton
+                                title={titleForStep}
+                                onBackButtonPress={navigateBack}
+                            />
+                            <TabSelector />
+                            {selectedTab === CONST.TAB.TAB_MANUAL ? (
```

Comment:
I don't really like creating Onyx keys for stuff like this. I also think it could be made more generic / reusable by passing in data for what tabs should be rendered.

My preference would be to refactor the `TabSelector` slightly so that instead of using Onyx it would be used like this:

```js
<TabSelector tabs={CONST.TAB}>
    {(selectedTab) => (
        {selectedTab === CONST.TAB.TAB_MANUAL && <MoneyRequestAmountPage />}
        {selectedTab === CONST.TAB.RECEIPT && <ReceiptSelector />}
    )}
</TabSelector>
```

Reply:
The selected tab value should be saved for the user (e.g. if they choose scan it should always open to scan), so it feels like we have to use Onyx, right?